src/pybind/mgr/prometheus/module.py

```diff
@@ -1562,6 +1562,20 @@ def get_collect_time_metrics(self) -> None:
                 cast(MetricCounter, sum_metric).add(duration, (method_name,))
                 cast(MetricCounter, count_metric).add(1, (method_name,))
 
+    def get_pg_repaired_objects(self):
+        dump = self.get('pg_dump')
+        for stats in dump['pg_stats']:
+            path = f'pg_objects_repaired{stats["pgid"]}'
+            self.metrics[path] = Metric(
+                    'counter',
+                    'pg_objects_repaired',
+                    'Number of objects repaired in a pg Count',
+                    ('pgid',)
```

Comment:
Do we really want this level of granularity? This might mean generating an extra 4k-16k metrics/scrape (@ljflores do we record PG count in the Telemetry stats? Any clue on per-cluster total PG count? Thanks!). Most of the time these metrics won't change, so the information per metric (entropy) is rather low. 

Alternatively, the total aggregated data (or per-pool at max) could be a better tradeoff between granularity and data efficiency (1 metric encodes pg repairs for a whole cluster or a pool).

For slow-changing metrics I'd rather follow this troubleshooting approach:
1. Users detect the issue in the aggregated metric via an alert.
2. If a user wanted to drill down and verify in which PG or OSD the repair happened, they could check logs?

Have you tried [to manually modify an object](https://docs.ceph.com/en/pacific/man/8/ceph-objectstore-tool/#manipulating-an-object-s-content) and see how the repair works and what traces it delivers?